### ISSUE TITLE
Roadmap for upcoming features/components?

### ISSUE THREAD

**Issue by Keepcase** [react-toolbox/react-toolbox#596]
Could we get a roadmap for upcoming features/components?
Something similar to https://github.com/callemall/material-ui/blob/master/ROADMAP.md

It'd be awesome if we could have some visibility on what's to come next!

Keep up the great work!

**Comment by vmsoftware** [comment]
@javivelasco, do you have some comment?

**Comment by javivelasco** [comment]
Hi guys!! So sorry for taking that long but I was thinking what would be the best. One of the things I wanted to accomplish with react-toolbox was to create a set of components as close as possible to the official MD spec. I would like to keep the repository with the components that are defined in the spec but in another issue we came up with the idea of creating a new repository with complementary components.

I think a Roadmap would fit pretty well in that new repository while this is just for the base MD component. In any case I'd start listing missing components from the MD spec and components that would be nice to have. Also another list with improvements that should be made to react-toolbox and plan it like a roadmap.

I'm not actually using RT in a project right now so it's more difficult for me to figure out missing stuff. We can open a document of something to gather ideas about next moves, components, improvements, etc. What do you guys think?

**Comment by javivelasco** [comment]
Closed in favor of #626

**Comment by vmsoftware** [comment]
@javivelasco 

Really a great idea. I think RT is the best MD library that I have seen never and congratulations for it. great work!. 

However I think the actual implementation of MD need some changes that would be excellent before think about new components or external components, I mean **responsive components**, a **12-columns based grid system** to organize the content and a **theme switcher** with 'light theme' and 'dark theme', proposed by MD specification.

I think those changes are more important at this moment and I'm going to suggest my personal opinion about how it should be:

For responsive components we only need to set the width of each component external wrapper to 100% and organize the internal content relative to this wrapper. You can say it's not usefull, but if you think about 12-columns based grid system really is the best option. Why?

let me try to explain it,

The 12-columns grid system, work as follow: 

1. Make a `.row` class with 100% of width.

2. Take 100% and divide between 12 then each columns have 8,3333333333% of width.

3. Make CSS classes for 1,2,3,4,5,6,7,8,9,10,11 and 12 columns like col1, col2, col3.... col12.  (8,3333333333% * column number). for example 

4. Make mediaQuerys for `.row` and `.colxx` classes if necessary.

```
.col3 {
 width: 25%;
}
```

Then the components with the 100%, fill `.colxx` wrapper and organize the components easy.

It looks like a hard work but if we take advantage of existing libraries it's really easy. 

I suggest [flexboxgrid.com](flexboxgrid.com). because it's ready to use and there is a react implementacion of this that use CSS Modules: [Flexboxgrid](https://github.com/roylee0704/react-flexbox-grid). Then I think is easy to merge with RT structure.

flexboxgrid include CSS classes for horizontal and vertical alignment, offset columns, distribution and it's really easy to use... the final user should use as follow:
```
<row>
   <col> any RT component</col>
   <col>another RT component</col>
</row>
...
```

About theme switcher, I think that is very useful and will reduce the issues about theming because the people will have a real example about how to customize the components (generally instead individually).

let me know what do you think
